test/xchainGauges/rootGaugeV2.ts

```diff
@@ -0,0 +1,302 @@
+import { Signer } from "ethers"
+import { deployments, ethers, network } from "hardhat"
+import {
+  AnyCallTranslator,
+  ArbitrumBridger,
+  GaugeController,
+  RootGaugeV2,
+  RootGaugeFactory,
+  SDL,
+  VotingEscrow,
+  Minter,
+} from "../../build/typechain"
+import { ANYCALL_ADDRESS, MULTISIG_ADDRESSES } from "../../utils/accounts"
+import { ALCHEMY_BASE_URL, CHAIN_ID } from "../../utils/network"
+import {
+  BIG_NUMBER_1E18,
+  convertGaugeNameToSalt,
+  getCurrentBlockTimestamp,
+  getWithName,
+  impersonateAccount,
+  setEtherBalance,
+  setTimestamp,
+} from "../testUtils"
+import { setupAnyCallTranslator, setupRootGaugeFactoryV2 } from "./utils"
+
+const { expect } = chai
+
+describe("RootGaugeV2", () => {
+  let signers: Array<Signer>
+  let users: string[]
+  let arbitrumBridger: ArbitrumBridger
+  let vesdl: VotingEscrow
+  let sdl: SDL
+  let rootGaugeFactory: RootGaugeFactory
+  let anyCallTranslator: AnyCallTranslator
+  let rootGauge: RootGaugeV2
+  let gaugeController: GaugeController
+  let minter: Minter
+
+  const GAS_LIMIT = 1000000
+  const GAS_PRICE = 990000000
+  const TEST_GAUGE_NAME = "testGauge"
+  const DAY = 86400
+  const WEEK = DAY * 7
+
+  const setupTest = deployments.createFixture(
+    async ({ deployments, ethers }) => {
+      await network.provider.request({
+        method: "hardhat_reset",
+        params: [
+          {
+            forking: {
+              jsonRpcUrl:
+                ALCHEMY_BASE_URL[CHAIN_ID.MAINNET] +
+                process.env.ALCHEMY_API_KEY,
+              blockNumber: 15542718,
+            },
+          },
+        ],
+      })
+
+      signers = await ethers.getSigners()
+      users = await Promise.all(
+        signers.map(async (signer) => signer.getAddress()),
+      )
+
+      // Get mainnet addresses
+
+      const sdlAddress = (await getWithName("SDL", "mainnet")).address
+      const veSDLAddress = (await getWithName("VotingEscrow", "mainnet"))
+        .address
+      const gaugeControllerAddress = (
+        await getWithName("GaugeController", "mainnet")
+      ).address
+      gaugeController = await ethers.getContractAt(
+        "GaugeController",
+        gaugeControllerAddress,
+      )
+
+      const minterAddress = (await getWithName("Minter", "mainnet")).address
+      sdl = await ethers.getContractAt("SDL", sdlAddress)
+      vesdl = await ethers.getContractAt("VotingEscrow", veSDLAddress)
+      minter = await ethers.getContractAt("Minter", minterAddress)
+
+      const contracts = await setupAnyCallTranslator(users[0], ANYCALL_ADDRESS)
+      anyCallTranslator = contracts.anyCallTranslator
+
+      // **** Setup rootGauge Factory ****
+      rootGaugeFactory = await setupRootGaugeFactoryV2(
+        anyCallTranslator.address,
+        users[0],
+        false,
+        sdlAddress,
+        gaugeControllerAddress,
+        minterAddress,
+      )
+      await anyCallTranslator.addKnownCallers([rootGaugeFactory.address])
```

Comment:
Since we can fork networks, we shouldnt manually recreate the environment like this unless we need to.
Forking latest block / epoch and using live addresses would be better in this case.